lib/Basic/Statistic.cpp

```diff
@@ -59,6 +56,16 @@ bool environmentVariableRequestedMaximumDeterminism() {
   return false;
 }
 
+uint64_t getInstructionsExecuted() {
+#if defined(HAVE_PROC_PID_RUSAGE) && defined(RUSAGE_INFO_V4)
+  struct rusage_info_v4 ru;
+  if (proc_pid_rusage(getpid(), RUSAGE_INFO_V4, (rusage_info_t *)&ru) == 0) {
+    return ru.ri_instructions;
+  }
+#endif
+  return 0;
```

Comment:
If we don't have rusage, `__builtin_readcyclecounter` emits a reasonable target-specific instruction stream that does this as well.

Reply:
After an offline discussion with Robert, we decided to stick with `rusage`'s `ri_instructions`. `ri_instructions`, `ri_cycles` and `__builtin_readcyclecounter` are giving different values in my tests and I’ve decided to stick with `ri_instructions` because that's what Graydon chose to use [here](https://github.com/apple/swift/pull/18658/files) and I trust he knew what he was doing (or at least knew more what he was doing than I do).